apps/generator/docs/generator-template.md

```diff
@@ -643,10 +643,558 @@ Temperature drop detected 66943992 sent to temperature/dropped
+        System.out.println("Validating generated generated Client.java");
+        System.out.println("Running tests in TestClient.java");
+        System.out.println("Sending temperature changes to the broker...");
+        System.err.println("\n");
+        while (true) {
+            int randomId = random.nextInt(maxValue - minValue + 1) + minValue;
+            client.sendTemperatureChange(String.valueOf(randomId));
+            System.out.println("New temperature detected " + randomId + " sent to temperature/changed");
+
+            try {
+                TimeUnit.SECONDS.sleep(1); // Sleep for 1 second
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+}
+```
+Run the command `gradle run -PmainClass=TestClient` to run your java program with gradle on your terminal. You should see output similar to the snippet below logged on your terminal:
+
+``` cmd
+New temperature detected 64250266 sent to temperature/changed
+New temperature detected 36947728 sent to temperature/changed
+New temperature detected 72955029 sent to temperature/changed
+```
+### 3. Output Java template code.
+Open [**index.js**](#java - indexjs-file), copy the content of [**client.py**](#1-create-the-client) and replace `{asyncapi.info().title()}` with it so it looks like the code snippet below now:
+
+```js
+//1
+import { File } from '@asyncapi/generator-react-sdk'
+//2
+export default function ({ asyncapi }) {
+//3
+  return <File name="Client.java">
+  {
+    `import org.eclipse.paho.client.mqttv3.*;
+
+public class Client {
+    private static final String BROKER_URL = "tcp://test.mosquitto.org:1883";
+    private static final String TOPIC = "temperature/changed";
+
+    private MqttClient client;
+
+    public Client() {
+        try {
+            // Generate a unique client ID
+            String clientId = MqttClient.generateClientId();
+                    
+            // Create and connect the MQTT client
+            client = new MqttClient(BROKER_URL, clientId);
+            MqttConnectOptions options = new MqttConnectOptions();
+            options.setCleanSession(true);
+                    
+            client.connect(options);
+            System.out.println("Connected to MQTT broker: " + BROKER_URL);
+        } catch (MqttException e) {
+            e.printStackTrace();
+        }
+    }
+     public void sendTemperatureChange(String id) {
+        try {
+            // Publish the message with the temperature change
+            MqttMessage message = new MqttMessage(id.getBytes());
+            client.publish(TOPIC, message);
+        } catch (MqttException e) {
+            e.printStackTrace();
+        }
+    }
+}`
+  }</File>
+}
+```
+##### b. Write the script to run the test code
+In **package.json** you can deine a scripts property that you invoke by calling `npm run <your_script>`. After adding these scripts in **package.json**, it will look like the following code snippet:
```

Comment:
```suggestion
In **package.json** define a script property that you invoke by calling `npm run <your_script>`. After adding these scripts in **package.json**, it will look like the following code snippet:
```